edge-detect/coral/sort_coral.py

```diff
@@ -0,0 +1,202 @@
+        self.mqtt_message_topic = mqtt_topic
+
+        # Make directories
+        os.makedirs(self.source_dir, exist_ok=True)
+        os.makedirs(self.plane_dir, exist_ok=True)
+        os.makedirs(self.noplane_dir, exist_ok=True)
+        os.makedirs(self.log_dir, exist_ok=True)
+
+        if len(self.labels) > 0:
+            labels = read_label_file(self.labels) if self.labels else {}
+            lookup = dict(zip(labels.values(), labels.keys()))
+            self.goal_index = lookup[self.goal]
+        else:
+            self.goal_index = 0
+        self.interpreter = make_interpreter(self.weights)
+        self.interpreter.allocate_tensors()
+        self.interpreter_input_size = common.input_size(self.interpreter)
+
+        self.connect_client()
+        time.sleep(1)
+        self.publish_registration("Coral Detection Module Registration")
+
+    def main(self: Any):
+        schedule.every(10).seconds.do(
+            self.publish_heartbeat, payload="Coral Detection Module Heartbeat"
+        )
+
+        while True:
+            try:
+                t = [time.time()]
+                while True:
+                    if os.path.isdir(self.source_dir):
+                        img_paths = glob.glob(os.path.join(self.source_dir, '*'))
+                        if len(img_paths) > 0:
+                            break
+                    time.sleep(self.delay_time)
+                img_path = min(img_paths, key=os.path.getctime)
+
+                # Load image
+                t.append(time.time())
+                image = Image.open(img_path)
+                image.thumbnail(self.interpreter_input_size) # Resize here
+                t.append(time.time())
+                _, scale = common.set_resized_input(
+                    self.interpreter, image.size, lambda size: image) # Trivial resize
+
+                # Inference
+                t.append(time.time())
+                self.interpreter.invoke()
+                t.append(time.time())
+                objs = detect.get_objects(self.interpreter, self.threshold, scale)
+                count = sum([obj.id == self.goal_index for obj in objs])
+
+                print("Inferring " + str(img_path))
+                print(objs)
+                print(count)
+
+                # Move image file
+                t.append(time.time())
+                if count > 0:
+                    print('plane')
+                    shutil.move(img_path, os.path.join(
+                        self.plane_dir, os.path.basename(img_path)))
+                else:
+                    print('noplane')
+                    shutil.move(img_path, os.path.join(
+                        self.noplane_dir, os.path.basename(img_path)))
+
+                # Write JSON
+                if self.save_json and count > 0:
+                    struct = []
+                    for obj in objs:
+                        if obj.id == self.goal_index:
+                            struct.append({
```

Comment:
@meadej  @dphogan - I think it makes sense to append some information about the model that ran the bounding box.  Can you add something in here with model type (TFLite), and some kind of version of the model in use?  Not sure what specifically that looks like

Reply:
Addressed in https://github.com/IQTLabs/edgetech-coral-detect/pull/1 and https://github.com/IQTLabs/edgetech-yolo-detect/pull/3